### ISSUE TITLE
[RN 0.52.0] Unknown named module: 'react'

### ISSUE THREAD

**Issue by mobdim** [bvic23/babel-plugin-functional-hmr#9]
Hello.

After upgrade react-native (0.52.0) I receive this error: `Unknown named module: 'react'`
![image](https://user-images.githubusercontent.com/483053/34771283-0d4460e0-f615-11e7-85d2-5ca14cca6848.png)

**Steps to Reproduce**
* Create new proj via  `react-native init`
* Install this plugin
* **Important!** Change file `App.js`:
```
const App = () => (
  <View style={styles.container}>
        <Text style={styles.welcome}>
          Welcome to React Native!
        </Text>
        <Text style={styles.instructions}>
          To get started, edit App.js
        </Text>
        <Text style={styles.instructions}>
          {instructions}
        </Text>
      </View>
)

export default App
```
Just run `react-native run-ios`

**Comment by bvic23** [comment]
Did you try to reset the cache and restart the packager? Something like this:
```
rm -rf node_modules/
yarn install
watchman watch-del-all
rm -rf $TMPDIR/react-*
```
and restart the packager.

**Comment by mobdim** [comment]
@bvic23 Yes, I've tried everything:

```
watchman watch-del-all
rm -rf $TMPDIR/react-*
rm -rf node_modules/
rm yarn.lock
rm package-lock.json
npm cache clean --force
yarn cache clean
yarn install
```

The only problem on RN 0.52.0 if you use RN 0.51.0, then everything is fine.

**Comment by bvic23** [comment]
I see :-(. Let me look into it and @mobdim feel free to send a PR if you find the solution earlier.

**Comment by fossage** [comment]
Same issue here.  Also on RN 0.52.0.

**Comment by onrige** [comment]
Also have this issue. It comes from metro bundler I think.

"If you have any scripts in your project that depend on metro-bundler, you will need to change those to metro ([related commit](https://github.com/facebook/metro/commit/0f7ad193c75183eeff1b356644ccf22b0813bb04)). A likely place for this is in rn-cli.config.js.

**Comment by Gyran** [comment]
A new version is now available that should close this
It was fixes in https://github.com/bvic23/babel-plugin-functional-hmr/pull/10

**Comment by bvic23** [comment]
Thanks to @Gyran!

**Comment by Barylskyigb** [comment]
Having same error after the fix on rn 0.52.2. Does anyone has same result?

in package.json, I have:
"babel-plugin-functional-hmr": "github:bvic23/babel-plugin-functional-hmr"

**Comment by Gyran** [comment]
Are you using Yarn? sure the latest version is downloaded?

**Comment by Barylskyigb** [comment]
@Gyran everything worked after I ran react-native start --reset-cache :) Thank you for the fix.

**Comment by anhnnp** [comment]
@Gyran @hbarylskyi thanks for much!
I ran react-native start --reset-cache and fixed hot reload on react native version 0.54
